src/store/cache/rewards/helpers.ts

```diff
@@ -0,0 +1,171 @@
+  const activeStake = totalStakedAtFundBlockForClaimer.sub(totalLocked)
+  const rewardsForClaimer = activeStake
+    .mul(fundingAmount)
+    .div(totalStakedAtFundBlock)
+  return rewardsForClaimer
+}
+
+export const getOperatorRewards = ({
+  user,
+  totalRewards,
+  deployerCutBase = DEPLOYER_CUT_BASE
+}: {
+  user: Operator
+  totalRewards: BN
+  deployerCutBase?: BN
+}) => {
+  const totalActiveStake = getOperatorTotalActiveStake(user)
+  const deployerCut = new BN(user.serviceProvider.deployerCut)
+
+  const totalDelegatedRewards = user.delegators.reduce((total, delegate) => {
+    const delegateRewards = getDelegateRewards({
+      delegateAmount: delegate.activeAmount,
+      totalRoundRewards: totalRewards,
+      totalActive: totalActiveStake,
+      deployerCut,
+      deployerCutBase
+    })
+    return total.add(delegateRewards.delegatorCut)
+  }, new BN('0'))
+
+  const operatorRewards = totalRewards.sub(totalDelegatedRewards)
+  return operatorRewards
+}
+
+export const getDelegateRewards = ({
+  delegateAmount,
+  totalRoundRewards,
+  totalActive,
+  deployerCut,
+  deployerCutBase = DEPLOYER_CUT_BASE
+}: {
+  delegateAmount: BN
+  totalRoundRewards: BN
+  totalActive: BN
+  deployerCut: BN
+  deployerCutBase?: BN
+}) => {
+  const rewardsPriorToSPCut = delegateAmount
+    .mul(totalRoundRewards)
+    .div(totalActive)
+  const spDeployerCut = delegateAmount
+    .mul(totalRoundRewards)
+    .mul(deployerCut)
+    .div(totalActive)
+    .div(deployerCutBase)
+  return {
+    spCut: spDeployerCut,
+    delegatorCut: rewardsPriorToSPCut.sub(spDeployerCut)
+  }
+}
+
+export const getRewardForClaimBlock = async ({
+  wallet,
+  users,
+  fundsPerRound,
+  blockNumber,
+  aud
+}: {
+  wallet: Address
+  users: (User | Operator)[]
+  fundsPerRound: BN
+  blockNumber: number
+  aud: AudiusClient
+}) => {
+  const user = users.find(u => u.wallet === wallet)
+  let totalRewards = new BN('0')
+  console.log('staging')
+  if ('serviceProvider' in user!) {
+    const lockedPendingDecrease =
+      (user as Operator).pendingDecreaseStakeRequest?.amount ?? new BN('0')
+    const lockedDelegation = await aud.Delegate.getTotalLockedDelegationForServiceProvider(
+      wallet
+    )
+    const totalLocked = lockedPendingDecrease.add(lockedDelegation)
+    const mintedRewards = await getMintedAmountAtBlock({
+      aud,
+      totalLocked,
+      fundingAmount: fundsPerRound,
+      wallet,
+      blockNumber
+    })
+    const operatorRewards = getOperatorRewards({
+      user: user as Operator,
+      totalRewards: mintedRewards
+    })
+    console.log(`Stake Adding ${operatorRewards.toString()}`)
+    totalRewards = totalRewards.add(operatorRewards)
+  }
+
+  for (let delegate of (user as User).delegates) {
+    const operator = users.find(u => u.wallet === delegate.wallet) as Operator
+    const deployerCut = new BN(operator.serviceProvider.deployerCut.toString())
```

Comment:
isn't deployerCut static throughout? it's the cut of the serviceProvider so we can probably calculate it once